src/components/PopupWithForm.js

```diff
@@ -22,11 +39,14 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
+      this.showLoading();
```

Comment:
Needs correction: you should start the loading in index.js right before the api request (because the form can be submitted using different ways)